staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -0,0 +1,264 @@
+// filter
+type requestManagement struct {
+	fairQueuingFactory FairQueuingFactory
+
+	// configQueue holds TypedConfigObjectReference values, identifying
+	// config objects that need to be processed
+	configQueue workqueue.RateLimitingInterface
+
+	// numConfigWorkers is the configured number of workers for the config queue
+	numConfigWorkers int
+
+	// plInformer is the informer for priority level config objects
+	plInformer cache.SharedIndexInformer
+
+	// fsInformer is the informer for flow schema config objects
+	fsInformer cache.SharedIndexInformer
+
+	// plLister belongs here too
+	// fsLister belongs here too
+
+	// serverConcurrencyLimit is the limit on the server's total
+	// number of non-exempt requests being served at once.  This comes
+	// from server configuration.
+	serverConcurrencyLimit int
+
+	// requestWaitLimit comes from server configuration.
+	requestWaitLimit time.Duration
+
+	// curState holds a pointer to the current RMState.  That is,
+	// `Load()` produces a `*RMState`.  When a config work queue worker
+	// processes a configuration change, it stores a new pointer here ---
+	// it does NOT side-effect the old `RMState` value.  The new `RMState`
+	// has a freshly constructed slice of FlowSchema pointers and a
+	// freshly constructed map of priority level states.  But the new
+	// `RMState.priorityLevelStates` includes in its range at least all the
+	// `*PriorityLevelState` values of the old `RMState.priorityLevelStates`.
+	// Consequently the filter can load a `*RMState` and work with it
+	// without concern for concurrent updates.  When a priority level is
+	// finally deleted, this will also involve storing a new `*RMState`
+	// pointer here, but in this case the range of the
+	// `RMState.priorityLevels` will be reduced --- by removal of the
+	// priority level that is no longer in use.
+	curState atomic.Value
+}
+
+// TypedConfigObjectReference is a reference to a relevant config API object.
+// No namespace is needed because none of these objects is namespaced.
+type TypedConfigObjectReference struct {
+	Kind string
+	Name string
+}
+
+func (tr *TypedConfigObjectReference) String() string {
+	return tr.Kind + "/" + tr.Name
+}
+
+// rmSetup is invoked at startup to create the infrastructure of this filter
+func rmSetup(loopbackClientConfig *restclient.Config, serverConcurrencyLimit int, requestWaitLimit time.Duration) *requestManagement {
+	reqMgmt := &requestManagement{
+		configQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue"),
+		serverConcurrencyLimit: serverConcurrencyLimit,
+		requestWaitLimit:       requestWaitLimit,
+	}
+	// TODO: finish implementation
+	return reqMgmt
+}
+
+// WithRequestManagement limits the number of in-flight requests in a fine-grained way
+func WithRequestManagement(
+	handler http.Handler,
+	loopbackClientConfig *restclient.Config,
+	serverConcurrencyLimit int,
```

Comment:
>      --max-mutating-requests-inflight int           The maximum number of mutating requests in flight at a given time. When the server exceeds this, it rejects requests. Zero for no limit. (default 200)
>      --max-requests-inflight int                    The maximum number of non-mutating requests in flight at a given time. When the server exceeds this, it rejects requests. Zero for no limit. (default 400)

iiuc, the limit value reads from these flags, amiright? `serverConcurrencyLimit` = `max-mutating-requests-inflight` + `max-requests-inflight`

Reply:
See https://github.com/kubernetes/enhancements/blob/79119ce549cda9cdfdb532273475ea5090fa5415/keps/sig-api-machinery/20190228-priority-and-fairness.md#primary-cpu-and-memory-protection , including its last paragraph.